Add `prefer-prototype-methods` rule (#1247)
sindresorhus/eslint-plugin-unicorn: docs/rules/prefer-prototype-methods.md

> # Prefer use methods from the prototype instead of methods from an instance.

Reviewer: This rule is an extension of the recommended eslint core rule [no-prototype-builtins](https://eslint.org/docs/rules/no-prototype-builtins) right? It would be good to link to that as a related rule in the rule doc.

Author: I don't think they are related.

> When "borrow" a method from different objects (especially generic methods from `Array`), it's more clear to use it from the constructor prototype.

Reviewer 2: ```suggestion
When “borrowing” a method from a different object (especially generic methods from `Array`), it‘s clearer to get it from the prototype than from an instance.
```
